pandas/core/arrays/masked.py

```diff
@@ -1661,7 +1661,9 @@ def transpose_homogeneous_masked_arrays(
     arr_type = dtype.construct_array_type()
     transposed_arrays: list[BaseMaskedArray] = []
     for i in range(transposed_values.shape[1]):
-        transposed_arr = arr_type(transposed_values[:, i], mask=transposed_masks[:, i])
```

Comment:
```
    masked_arrays = list(masked_arrays)
    dtype = masked_arrays[0].dtype

    values = [arr._data.reshape(1, -1) for arr in masked_arrays]
    transposed_values = np.concatenate(values, axis=0, out=np.empty((len(masked_arrays), len(masked_arrays[0])), order="F", dtype=dtype.numpy_dtype))

    masks = [arr._mask.reshape(1, -1) for arr in masked_arrays]
    transposed_masks = np.concatenate(masks, axis=0, out=np.empty_like(transposed_values, dtype=np.bool))

    arr_type = dtype.construct_array_type()
    transposed_arrays: list[BaseMaskedArray] = []
    for i in range(transposed_values.shape[1]):
        transposed_arr = arr_type(
            transposed_values[:, i], mask=transposed_masks[:, i]
        )
        transposed_arrays.append(transposed_arr)

    return transposed_arrays
```

this should be faster (avoids the second copy, but not sure if I can come up with a meaningful benchmark...)


```
arr = [
    pd.array(np.random.randint(1, 1_000_000, (100, )), dtype="Int64"),
    pd.array(np.random.randint(1, 1_000_000, (100, )), dtype="Int64"),
    pd.array(np.random.randint(1, 1_000_000, (100, )), dtype="Int64"),
    pd.array(np.random.randint(1, 1_000_000, (100, )), dtype="Int64"),
]

arr = arr * 10_000

transpose_homogeneous_masked_arrays(arr)
```

Reply:
Nice!

```
size = 100_000
df = pd.DataFrame(
    {
        "a": pd.array([0]*size, dtype="Int8"),
        "b": pd.array([0]*size, dtype="Int8"),
    }
)
%timeit df.T

# main
416 ms ± 1.43 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

# orig PR
477 ms ± 4.38 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

# phofl
417 ms ± 1.98 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

Reply:
Ah nice that we keep consistent on the Python dominated benchmark as well